shell/platform/android/io/flutter/embedding/engine/systemchannels/SettingsChannel.java

```diff
@@ -80,7 +105,31 @@ public void send() {
               + "\n"
               + "platformBrightness: "
               + message.get(PLATFORM_BRIGHTNESS));
-      channel.send(message);
+      final DisplayMetrics metrics = this.displayMetrics;
+      if (Build.VERSION.SDK_INT < 34 || metrics == null) {
```

Comment:
If we guard this by SDK version, should we then also throw if `getPastDisplayMetrics` is accessed in an unsupported android version since it unexpectedly wouldn't return you any metrics.

Reply:
Let me see if there's an annotation for that.